COLLABORATOR_GUIDE.md

```diff
@@ -84,6 +87,191 @@ All pull requests that modify executable code should be subjected to
 continuous integration tests on the
 [project CI server](https://ci.nodejs.org/).
 
+### Internal vs. Public API
+
+Due to the nature of the JavaScript language, it can often be difficult to
+establish a clear distinction between which parts of the Node.js implementation
+represent the "public" API Node.js users should assume to be stable and which
+are considered part of the "internal" implementation detail of Node.js itself.
+A general rule of thumb has been to base the determination off what
+functionality is actually *documented* in the official Node.js API
+documentation. However, it has been repeatedly demonstrated that either the
+documentation does not completely cover implemented behavior or that Node.js
+users have come to rely heavily on non-documented aspects of the Node.js
```

Comment:
`undocumented` or `not documented`, please.

`non-documented` is not in Merriam-Webster, OED, Wordnik, Wiktionary...

(For the record, Wikitionary does have `nondocumented` without the hyphen, but showing up in Wiktionary is probably the least convincing of those four sources. It's like not being in a few different published encyclopedias but showing up in Wikipedia. It doesn't count for *nothing*, but by itself....eh....)